credential_provider/credential_provider.go

```diff
@@ -1,18 +1,13 @@
 package credential_provider
 
 import (
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/aws/credentials"
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
 )
 
-// CredentialProvider interface defines methods for obtaining AWS credentials configuration
-type CredentialProvider interface {
+// ConfigProvider interface defines methods for obtaining AWS credentials configuration
+type ConfigProvider interface {
 	// GetAWSConfig returns an AWS configuration containing credentials obtained from the provider
-	GetAWSConfig() (*aws.Config, error)
-}
-
-// authTokenFetcher interface defines methods for fetching a token given a K8s namespace and service account.
-// It matches stscreds.TokenFetcher interface.
-type authTokenFetcher interface {
-	FetchToken(ctx credentials.Context) ([]byte, error)
+	GetAWSConfig(ctx context.Context) (aws.Config, error)
```

Comment:
Rather than force a unifying method signature for getting a token and creating unnecessary wrappers, we can use the interface types provided in each credential method for retrieving tokens, and just make the `ConfigProvider` return an `aws.Config{}` with the correct AWS cred provider

* For IRSA, [we can use `stscreds.IdentityTokenRetriever`](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/credentials@v1.17.59/stscreds#IdentityTokenRetriever)
* For Pod Identity, [we can use `endpointcreds.AuthTokenProvider`](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/credentials@v1.17.59/endpointcreds#AuthTokenProvider)